spec/workers/commit_monitor_handlers/commit_range/github_pr_commenter/commit_metadata_checker_spec.rb

```diff
@@ -0,0 +1,117 @@
+describe CommitMonitorHandlers::CommitRange::GithubPrCommenter::CommitMetadataChecker do
+  let(:batch_entry)        { BatchEntry.create!(:job => BatchJob.create!) }
+  let(:branch)             { create(:pr_branch) }
+  let(:commit_message_1)   { "Adds SUPER FEATURE.  BREAKS NOTHING!" }
+  let(:commit_message_2)   { "fix tests" }
+  let(:commit_message_3)   { "fix moar tests" }
+
+  let(:commits) do
+    {
+      "abcd123" => {"message" => commit_message_1, "files" => ["app/models/super.rb"]},
+      "abcd234" => {"message" => commit_message_2, "files" => ["spec/models/super_spec.rb"]},
+      "abcd345" => {"message" => commit_message_3, "files" => ["spec/models/normal_spec.rb"]}
+    }
+  end
+
+  let(:username_lookup_cache) do
+    {
+      "Fryguy"          => 123, # valid user
+      "NickLaMuro"      => 234, # valid user
+      "chrisarcand"     => 345, # valid user
+      "NickLaMura"      => nil, # mispelled invalid user
+      "Nick-LaMuro"     => nil, # invalid user with hyphen
+      "booksandauthors" => nil  # not a user (surprisingly)
+    }
+  end
+
+  before do
+    stub_sidekiq_logger
+    stub_job_completion
+
+    allow(GithubService).to receive(:username_lookup_cache).and_return(username_lookup_cache)
+  end
+
+  context "with basic commit messages" do
+    it "doesn't create any offenses" do
+      described_class.new.perform(batch_entry.id, branch.id, commits)
+
+      batch_entry.reload
+      expect(batch_entry.result.length).to eq(0)
+    end
+  end
+
+  context "with multiline commit messages" do
+    let(:commit_message_1) do
+      <<~COMMIT_MSG
+        In which the hero adds the greatest feature...
+
+        With this change, I made it so that by changing this:
+
+            @database_records = Books.all
+
+        Avoids a N+1 on authors by adding an .includes call
+
+            @booksandauthors = Books.all.includes(:author)
+
+        Saving us 100s of queries for our tiny bookstore app.
+
+        Thanks to @Fryguy and @Nick-LaMuro for this suggestion!
+      COMMIT_MSG
+    end
+
+    let(:commit_message_2) do
+      <<~COMMIT_MSG
+        fixes tests
+
+        Forgot that we stubbed things...
+
+        cc @Fryguy @NickLaMuro
+      COMMIT_MSG
+    end
+
+    let(:commit_message_3) do
+      <<~COMMIT_MSG
+        fixes moar tests
+
+        I forget how to rebase... #dealWithIt @NickLaMura
+      COMMIT_MSG
+    end
+
+    it "returns one offense for each valid username" do
+      described_class.new.perform(batch_entry.id, branch.id, commits)
+
+      batch_entry.reload
+      expect(batch_entry.result.length).to eq(3)
+      expect(batch_entry.result.first).to have_attributes(
+        :group   => "abcd123",
+        :message => "Username `@Fryguy` detected in commit message. Consider removing."
```

Comment:
Reading this, it might be helpful to give a reference to the commit in question...

    Username `@Fryguy` detected in commit message for commit a0d8f20d. Consider removing.

Perhaps even link the commit like we do in other error messages.

Reply:
So pretty sure that will happen with `group`, which is what I was doing here.

Instead of the filename here:

https://github.com/ManageIQ/miq_bot/pull/426#issuecomment-432861521

It would just print the commit SHA, which should auto link based on the autolink spec:

https://help.github.com/en/github/writing-on-github/autolinked-references-and-urls#commit-shas

That said, I might need to add it as `"#{issuer}@#{sha}"`... unsure.

Reply:
Just found out that this is a thing:

https://github.com/ManageIQ/miq_bot/blob/d5a8b5e00895c9fa449a42ca317801e4bbef8c07/app/models/branch.rb#L25-L27

So I will make use of that for the group, which should solve your concern.